shared/src/main/scala/mlscript/codegen/typescript/TsTypegen.scala

```diff
@@ -122,38 +113,22 @@ final class TsTypegenCodeBuilder {
       .filter(tup => methodBodyType.freeTypeVariables.contains(tup._1) &&
         // ignore class type parameters since they are implicitly part of class scope
         // if no name hint then the type variable is certainly not a class type parameter
-        // it's friendly name has been generated
+        // its friendly name has been generated
         tup._1.nameHint.fold(true)(!classTypeParams.contains(_))
       )
       .map(_._2)
       .toList
 
     val methodSourceCode = methodBodyType match {
       // function creates signature with `:` separating arguments and body
```

Comment:
I don't understand this. Where is the `:`? Can you give examples of the two cases and what they look like?

Reply:
```
method Unlock: Lock[T] -- constant value
method Map: (T -> 'a) -> Lock['a] -- function
```

In the case of the function the type generation will give `(arg1: (arg3: T) => a)":" Lock<a>`. The ":" is added by the type generation itself.

However in the first case where the method is a constant value that ":" is never added so it has to be handled as a separate case.

Reply:
Ok, but that seems pretty brittle. You're basically making a big assumption on what another part of the code far away will do. If any of these two parts change, the assumption could be broken. And it does not seem to be documented anywhere (I'd expect it in the doc of `toTsType`?).

Reply:
BTW, for the case without parameter lists, I think you should generate a `readonly x: T` declaration, and raise an error if the definition has type parameters (not supported by TS).

Reply:
The new approach we discussed removes the need for handling special case for method type generation in the `toTsType`. Now it does the handling in `addClassMethod` itself. So there is no need for a `isMethodDefinition` variable to change the source code representation.

Also known value methods are readonly attributes and type variables are not allowed for readonly attributes.